static/app/components/profiling/textTruncateOverflow.tsx

```diff
@@ -0,0 +1,24 @@
+import {ElementType} from 'react';
+import styled from '@emotion/styled';
+
+import space from 'sentry/styles/space';
+
+interface TextTruncateOverflowProps {
+  children: string;
+  as?: ElementType<any>;
+}
+
+// TextTruncateOverflow is strictly a css based text truncate component
+export const TextTruncateOverflow = styled(
+  ({children, as: Element = 'div', ...props}: TextTruncateOverflowProps) => (
```

Comment:
Does this infer correct props depending on the Element you pass?

Reply:
Since i've supplied in a component, i'd say its probably unlikely. I can add that in.

Reply:
i fixed it up so it does now... 
![image](https://user-images.githubusercontent.com/7349258/215509383-fe33896e-3e6b-49e1-87cb-8f1f2d0b93a5.png)